### PR TITLE
bisonpp: init at 1.21-45

### PR DESCRIPTION
**Pull request by costrouc**
Adding bisonpp library (no tests included)

###### Motivation for this change

Needed for building https://github.com/omnisci/omniscidb

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by costrouc** [comment]
@GrahamcOfBorg build bisonpp

File: pkgs/development/tools/parsing/bisonpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchFromGitHub {
+    owner = "jarro2783";
+    repo = "bisonpp";
+    rev = version;
+    sha256 = "0mp6cvbm2rfjwff3n9xz0gfxmdamad55nlr9f5q5dy0h6c43mh8x";
+  };
+
+  meta = with stdenv.lib; {
+    description = "The original bison++ project, brought up to date with modern compilers";
+    homepage = https://github.com/jarro2783/bisonpp;
```
Comment (Valodim) [inline pkgs/development/tools/parsing/bisonpp/default.nix]:
```suggestion
    homepage = "https://github.com/jarro2783/bisonpp";
````

**Review by Valodim — CHANGES_REQUESTED** [review]
lgtm, works as expected, other than the uri comment and merge conflict.

**Comment by onny** [comment]
Please rebase

**Pull request closed without merging**